### PR TITLE
Add support to Symfony file copier for exclusions

### PR DESCRIPTION
**Pull request by TravisCarden**
...and fix directory recursion. 😬

### CODE REVIEW

File: src/Infrastructure/Process/FileCopier/SymfonyFileCopier.php [hunk 1]
```diff
@@ -39,34 +43,41 @@ public function copy(
...
-        /** @var callable(mixed, mixed, \RecursiveIterator<mixed, mixed>):bool $callback */
-        $callback = function (SplFileInfo $current): bool {
-            // Ignore current and parent directories.
-            if (in_array($current->getFilename(), ['.', '..'], true)) {
-                return false;
-            }
+            $this->finder
+                ->in($from)
+                ->filter(function (SplFileInfo $current) use ($from, $exclusions): bool {
+                    // Make path name relative to "from" path like exclusions.
```
Comment (phenaproxima) [inline src/Infrastructure/Process/FileCopier/SymfonyFileCopier.php:64]: Do we need to use a callback here? Finder has a lot of powerful built-in filtering, like `notPath()` and things like that. I don't object to using the callback; I'm just wondering if we're reinventing the wheel a bit. If we're not, maybe a comment would be handy to explain what the callback is doing that precludes the use of Finder's features.

File: src/Infrastructure/Process/FileCopier/SymfonyFileCopier.php [hunk 2]
```diff
@@ -39,34 +43,41 @@ public function copy(
...
+                    $pathName = $current->getPathname();
+                    if (strpos($pathName, $from) === 0) {
+                        // Strip the "from" path from the beginning of the path.
+                        $pathName = substr($pathName, strlen($from));
+                        $pathName = ltrim($pathName, DIRECTORY_SEPARATOR);
+                    }
 
-            return true;
-        };
+                    if (in_array($pathName, $exclusions, true)) {
```
Comment (phenaproxima) [inline src/Infrastructure/Process/FileCopier/SymfonyFileCopier.php:72]: Nit: Couldn't we just `return !in_array(...)`?

**Review by phenaproxima — APPROVED** [review]

File: src/Infrastructure/Process/FileCopier/SymfonyFileCopier.php (hunk 1 above)
Comment (TravisCarden) [reply src/Infrastructure/Process/FileCopier/SymfonyFileCopier.php:64]: _Ooh..._ Good call, @phenaproxima -- that simplifies things significantly!

File: tests/Unit/Infrastructure/Process/FileCopier/SymfonyFileCopierTest.php [hunk 3]
```diff
@@ -74,8 +83,21 @@ public function providerCopy(): array
...
+     * @uses \PhpTuf\ComposerStager\Exception\DirectoryNotFoundException
+     * @uses \PhpTuf\ComposerStager\Exception\PathException
      */
+    public function testCopyDirectoryNotFound(): void
+    {
+        $this->expectException(DirectoryNotFoundException::class);
+
+        $this->finder
+            ->in(Argument::any())
+            ->willThrow(\Symfony\Component\Finder\Exception\DirectoryNotFoundException::class);
```
[the remark this answers is not in the document]
Comment (TravisCarden) [reply tests/Unit/Infrastructure/Process/FileCopier/SymfonyFileCopierTest.php:95]: ...except that you lose certain IDE features if you use a string. Also, I just discovered, testing it out, that PhpStorm actually flags it if you do. (Interesting.)

**Pull request merged**